### ISSUE TITLE
Order creation and watching

### ISSUE THREAD

**Issue by wunderb1t358** [ccxt/ccxt#8245]
Hi! How to correctly organize order creation and further order watching? Please see code below. 

P.S. What value has timeout after which socket connection attempt is considered as unsuccessful?

- OS: DEBIAN stretch (Docker)
- Programming Language version: NodeJS (v13.12.0)
- CCXT version: Pro version 0.5.20

```
// Function to watch created orders.
async function monitorOrders() {
    const exchange = new ccxt.binance({
        apiKey: 'apiKey',
        secret: 'secret'
    });

    //real-time mode
    while (true) {
        try {
            const orders = await exchange.watchOrders();
            console.log (orders); // every update
        } catch (e) {
            console.log (e);
            //throw e // uncomment to stop the loop on exceptions
        }
    }
}

// Function to create orders.
async function createOrder() {
    const exchange = new ccxt.binance({
        apiKey: 'apiKey',
        secret: 'secret'
    });

    await exchange.createOrder(...);
}

// The problem code.
async function app() {
    await monitorOrders();

    // THIS CODE WILL NOT BE REACHED, because there are no any orders.
    // watchOrders promise can't be resolved without new orders.
    await createOrder();
}

// Possible solution.
function app() {
    monitorOrders();
    
    // First give watchOrders time to establish connection, then create order.
    // !!! BUT what to do if watchOrders has connection delay more than 2s and result will be exception?
    // Created orders can be missed by watchOrders...
    setTimeout(() => {
        createOrder();
    }, 2000);
}
```

**Comment by kroitor** [comment]
Hi @wunderb1t358 ! We will post an example here shortly.

**Comment by kroitor** [comment]
Binance WebSocket API starts streaming private feeds when the connection is established.
Therefore `watch_orders()` will not return a value until the first order update is received from the exchange.

We have added a base method that you can use to hook-up to the on_connected event.
It is available in CCXT Pro version 0.5.30, so you might need to update to the most recent version.

Let us know if this example does not resolve your issue after upgrading to CCXT Pro 0.5.30:

- https://github.com/ccxt/ccxt/tree/master/examples/ccxt.pro/py/on-connected-user-hook.py


```Python
import ccxtpro
from asyncio import get_event_loop, ensure_future
from pprint import pprint


print('CCXT Pro Version:', ccxtpro.__version__)


# on_connected() is called when a client connection is established
# note that the exchange will reuse the same client connection
# some exchanges might require two or more public/private connections
# therefore on_connected() may be called more than once

class MyBinance(ccxtpro.binance):
    def on_connected(self, client, message):
        print('Connected to', client.url)
        ensure_future(create_order(self))


async def create_order(exchange):
    symbol = 'BTC/USDT'
    type = 'limit'
    side = 'buy'
    amount = 123.45  # change for your values
    price = 54.321  # change for your values
    params = {}
    try:
        order = await exchange.create_order(symbol, type, side, amount, price, params)
        print('--------------------------------------------------------------')
        print('create_order():')
        pprint(order)
    except Exception as e:
        print(type(e).__name__, str(e))
        break


async def watch_orders(exchange):
    while True:
        try:
            orders = await exchange.watch_orders()
            print('--------------------------------------------------------------')
            print('watch_orders():')
            pprint(orders)
        except Exception as e:
            print(type(e).__name__, str(e))
            break;
    await exchange.close()


loop = get_event_loop()
exchange = MyBinance({
    'enableRateLimit': True,
    'apiKey': 'YOUR_API_KEY',
    'secret': 'YOUR_SECRET',
    'asyncio_loop': loop,
})

loop.run_until_complete(watch_orders(exchange))
```

**Comment by emacasa** [comment]
I tried to run this exact code, just substituting Binance with Poloniex, and this error happens. Is this normal?

Thanks

Exception in callback FastClient.receive_loop.<locals>.handler() at /Users/.../venv/lib/python3.8/site-packages/ccxtpro/base/fast_client.py:20
handle: <Handle FastClient.receive_loop.<locals>.handler() at /Users/.../venv/lib/python3.8/site-packages/ccxtpro/base/fast_client.py:20>
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/events.py", line 81, in _run
    self._context.run(self._callback, *self._args)
  File "/Users/.../venv/lib/python3.8/site-packages/ccxtpro/base/fast_client.py", line 25, in handler
    self.handle_message(message)
  File "/Users/.../venv/lib/python3.8/site-packages/ccxtpro/base/aiohttp_client.py", line 32, in handle_message
    self.handle_text_or_binary_message(message.data)
  File "/Users/.../venv/lib/python3.8/site-packages/ccxtpro/base/aiohttp_client.py", line 27, in handle_text_or_binary_message
    self.on_message_callback(self, decoded)
  File "/Users/.../venv/lib/python3.8/site-packages/ccxtpro/poloniex.py", line 776, in handle_message
    method(client, message)
  File "/Users/.../venv/lib/python3.8/site-packages/ccxtpro/poloniex.py", line 437, in handle_account_notifications
    callback(client, entry)
  File "/Users/.../venv/lib/python3.8/site-packages/ccxtpro/poloniex.py", line 465, in handle_balance
    fetchedBalance = self.safe_value(subscription, 'fetchedBalance', False)
  File "/Users/.../venv/lib/python3.8/site-packages/ccxt/base/exchange.py", line 749, in safe_value
    return dictionary[key] if Exchange.key_exists(dictionary, key) else default_value
  File "/Users/.../venv/lib/python3.8/site-packages/ccxt/base/exchange.py", line 689, in key_exists
    if key in dictionary:
TypeError: argument of type 'bool' is not iterable
